src/js/api/core-loader.js

```diff
@@ -1,190 +1,84 @@
-    if (!controllerPromise) {
-        controllerPromise = new Promise(function (resolve) {
-            require.ensure(['controller/controller'], function (require) {
-                const CoreMixin = require('controller/controller');
-                resolve(CoreMixin);
-            }, 'jwplayer.core');
-        });
+export default function loadCoreBundle(model) {
+    if (!bundlePromise) {
+        bundlePromise = selectBundle(model);
     }
-    return controllerPromise;
+    return bundlePromise;
 }
 
-const CoreModel = function() {};
-Object.assign(CoreModel.prototype, SimpleModel);
-
-const CoreLoader = function CoreSetup(originalContainer) {
-    loadController();
-    this._events = {};
-    this.controller = null;
-    this.model = new CoreModel();
-    this.model._qoeItem = new Timer();
-    this.originalContainer = originalContainer;
-    this.apiQueue = new ApiQueueDecorator(this, [
-        // These commands require a provider instance to be available
-        'load',
-        'play',
-        'pause',
-        'seek',
-        'stop',
-        'playlistItem',
-        'playlistNext',
-        'playlistPrev',
-        'next',
-
-        // These should just update state that could be acted on later, but need to be queued given v7 model
-        'setConfig',
-        'setCurrentAudioTrack',
-        'setCurrentCaptions',
-        'setCurrentQuality',
-        'setFullscreen',
-        'addButton',
-        'removeButton',
-        'castToggle',
-        'setMute',
-        'setVolume',
-        'setPlaybackRate',
-
-        // These commands require the view instance to be available
-        'resize',
-        'setCaptions',
-        'setControls',
-        'setCues',
-    ], () => true);
-};
-
-Object.assign(CoreLoader.prototype, {
-    on: Events.on,
-    once: Events.once,
-    off: Events.off,
-    trigger: Events.trigger,
-    init(options, api) {
+function selectBundle(model) {
+    const controls = model.get('controls');
+    const polyfills = !('IntersectionObserver' in window &&
+        'IntersectionObserverEntry' in window &&
+        'intersectionRatio' in window.IntersectionObserverEntry.prototype);
+    const html5Provider = firstItemHandledByHtml5(model);
 
-        const storage = new Storage('jwplayer', [
-            'volume',
-            'mute',
-            'captionLabel',
-            'qualityLabel'
-        ]);
-        const persisted = storage && storage.getAllItems();
-        this.model.attributes = this.model.attributes || {};
-        Object.assign(this.model.attributes, new Config(options, persisted), playerDefaults);
+    if (controls && html5Provider) {
+        return loadControlsHtml5Bundle();
+    }
+    if (controls && polyfills && html5Provider) {
+        return loadControlsPolyfillHtml5Bundle();
+    }
+    if (controls && polyfills) {
+        return loadControlsPolyfillBundle();
+    }
+    if (controls) {
+        return loadControlsBundle();
+    }
+    return loadCore();
+}
```

Comment:
Couldn't these loaders just be one method that that take in an array. Think it would cleaner to read. For exampl

```
let name;
let bundles;
...
if (controls && polyfils) {
  bundles = [....];
  name = 'jwplayer.core.controls.polyfills';
}
...
loadBundles(name, bundles);
```